packages/z/zbar/port/xmake.lua

```diff
@@ -0,0 +1,120 @@
+add_rules("mode.debug", "mode.release")
+
+add_requires("libiconv")
+
+-- add options
+option("symbologies")
+    -- set_default({"ean","databar","code128","code93","code39","codabar","i25","qrcode","sqcode"})
+    set_description("Select symbologies to compile")
+option_end()
+
+option("vers")
+    set_default("")
+    set_showmenu(true)
+option_end()
+if has_config("vers") then
+    set_configvar("VERSION", get_config("vers"))
+    set_configvar("PACKAGE_VERSION", get_config("vers"))
+    set_configvar("PACKAGE_STRING", "zbar " .. get_config("vers"))
+    
+    local vers = get_config("vers"):split("%.")
+    major_ver = vers[1] or ""
+    minor_ver = vers[2] or ""
+    patch_ver = vers[3] or ""
+    set_configvar("ZBAR_VERSION_MAJOR", major_ver, {quote = false})
+    set_configvar("ZBAR_VERSION_MINOR", minor_ver, {quote = false})
+    set_configvar("ZBAR_VERSION_PATCH", patch_ver, {quote = false})
```

Comment:
use set_versions, xmake has builtin vars in config.h.in

see https://github.com/tboox/tbox/blob/1c3c01b1757febc4ade5ae16ce6a90c624b07d3c/src/tbox/tbox.config.h.in#L5-L9

Reply:
The `config.h.in` file contains content like the following:
```
/* Program major version (before the '.') as a number */
#undef ZBAR_VERSION_MAJOR

/* Program minor version (after '.') as a number */
#undef ZBAR_VERSION_MINOR

/* Program patch version (after the second '.') as a number */
#undef ZBAR_VERSION_PATCH
```
I'm not sure how to handle this.

Reply:
see https://github.com/tboox/tbox/blob/1c3c01b1757febc4ade5ae16ce6a90c624b07d3c/src/tbox/tbox.config.h.in#L5-L9

Reply:
Should I edit `config.h.in` to the following?
```
/* Program major version (before the '.') as a number */
#define ZBAR_VERSION_MAJOR ${VERSION_MAJOR}

/* Program minor version (after '.') as a number */
#define ZBAR_VERSION_MINOR ${VERSION_MINOR}

/* Program patch version (after the second '.') as a number */
#define ZBAR_VERSION_PATCH ${VERSION_ALTER}
```

Reply:
right